mockwebserver-deprecated/build.gradle.kts

```diff
@@ -15,7 +15,7 @@ tasks.jar {
 }
 
 dependencies {
-  api(projects.okhttp)
+  "friends"(projects.okhttp)
```

Comment:
Seems like this requires two configurations: `implementationFriend` and `apiFriend`. Otherwise you are changing the visibility of the transitive dependency, as evidence by all the unrelated changes below where you are having to add the dependency yourself when it should be automatically available.